src/test/java/org/terasology/launcher/game/UnixProcesses.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2020 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.launcher.game;
+
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Random;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+public final class UnixProcesses {
+    static final Callable<Process> COMPLETES_SUCCESSFULLY = runProcess("true");
+    static final Callable<Process> COMPLETES_WITH_ERROR = runProcess("false");
+    static final Callable<Process> NO_SUCH_COMMAND = runProcess(() -> {
+        // If you have a program with this name on your path while running these tests,
+        // you have incredible luck.
+        return "nope" + new Random()
+            .ints(16, 0, 255).mapToObj(
+                i -> Integer.toString(i, Character.MAX_RADIX)
+            )
+            .collect(Collectors.joining());
```

Comment:
Maybe less custom coding here?
```suggestion
        return "nope-" + UUID.randomUUID().toString();
```

Reply:
oh, this was just me making up garbage names for testing files? Yeah, sure, UUID is fine.

You get less entropy-per-character in UUID.toString() because it's formatted as base-16 and I think even type-4 UUIDs have at least a couple bits that aren't actually random, but I can't really defend the claim that those details are important to this use case.